### PR TITLE
Vm details screen

### PR DESCRIPTION
**Pull request by chalettu**
This PR is to introduce a VM Detail screen that is able to be reached from the service detail page.

### CODE REVIEW

**Comment by abellotti** [comment]
Nice job @chalettu on the API call made for this and all the attributes gathered.

**Comment by chalettu** [comment]
@miq-bot add_label enhancement

**Comment by chriskacerguis** [comment]
@chalettu can you address the test failure please?

**Comment by chalettu** [comment]
I saw that issue. I am taking a look at what is causing it.

File: client/app/states/vms/details/details.html [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+		</tr>
+		<tr>
+			<td>{{'Hostnames'|translate}}</td>
+			<td>{{vm.vmDetails.hostnames.join(",")}}</td>
+		</tr>
+		<tr>
+			<td>{{'IP Addresses'|translate}}</td>
+			<td>{{vm.vmDetails.ipaddresses.join(",")}}</td>
+		</tr>
+		<tr ng-show="vm.vmDetails.instance">
```
Comment (AllenBW) [inline client/app/states/vms/details/details.html]: prefer ng-if to ng-show (most always)

File: client/app/states/vms/details/details.html [hunk 2]
```diff
@@ -0,0 +1,513 @@
+<ol class="breadcrumb">
+  <li>
+    <a ui-sref="services">{{'My Services'|translate}}</a>
+  </li>
+  <li>
+    <a ui-sref="services.details({serviceId: vm.vmDetails.service.id})">{{ ::vm.vmDetails.service.name }}</a>
+  </li>
+  <li>
+    <a ui-sref="vms.details({vmId: vm.vmDetails.id})">{{ ::vm.vmDetails.name }}</a>
```
Comment (AllenBW) [inline client/app/states/vms/details/details.html]: YAY SINGLE BINDING! any reason why the rest of this isn't? are we polling the collection?  (if not, and there isn't a chance it'll be updated, all in this should be single bound, colon colon **FOREVER**)

File: client/app/states/vms/details/_details.sass [hunk 3]
```diff
@@ -0,0 +1,25 @@
+.list-view-container
+  &.order-details
+    height: calc(100vh - 272px)
+
+.ss-details-wrapper
+  .order-details
+    .list-view-pf
+      margin-top: 0
+
+$vm-detail-border-color: #d1d1d1;
```
Comment (AllenBW) [inline client/app/states/vms/details/_details.sass]: So a great time to get familiar with _colors.sass, declaring the hex code here is a nono (furthermore this color already exists) (makes reskinning a MASSIVE *B*) SO check it out, see what the color name should be, (hint `$color...`)

File: client/app/states/vms/details/details.html [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+			<td>
+				<i class="fa fa-camera fa-lg" aria-hidden="true"></i>{{vm.vmDetails.snapshotCount}}
+			</td>
+		</tr>
+		<tr>
+			<td>
+				{{'Advanced Settings'|translate}}
+			</td>
+			<td>
+				<i class="fa fa-wrench fa-lg" aria-hidden="true"></i>{{vm.vmDetails.advanced_settings.length}}
```
Comment (AllenBW) [inline client/app/states/vms/details/details.html]: we don't (presently) but should (some day) use aria, no reason to specify aria-hidden here

File: client/app/states/vms/vms.state.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  angular.module('app.states')
+    .run(appRun);
+
+  /** @ngInject */
+  function appRun(routerHelper) {
+    routerHelper.configureStates(getStates());
+  }
+
+  function getStates() {
+    return {
```
Comment (AllenBW) [inline client/app/states/vms/vms.state.js]: it this is a parent state, it should redirect somewhere, a vm list or explorer or something

File: client/app/states/vms/details/details.state.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    };
+
+    return CollectionsApi.get('vms', $stateParams.vmId, options);
+  }
+  /** @ngInject */
+  function StateController($state, vmDetails, CollectionsApi, EventNotifications, Consoles) {
+    var vm = this;
+  
+    vm.vmDetails = vmDetails;
+    setInitialVars();
```
Comment (AllenBW) [inline client/app/states/vms/details/details.state.js]: convention used throughout the app is to call this `activate()`

File: client/app/states/vms/details/details.state.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+      vm.vmDetails.snapshotCount = defaultText(vm.vmDetails.snapshots);
+      vm.vmDetails.resourceAvailability = (vm.vmDetails.template === false ? availableText : noneText);
+      vm.vmDetails.driftHistory = defaultText(vm.vmDetails.drift_states);
+      vm.vmDetails.scanHistoryCount = defaultText(vm.vmDetails.scan_histories);
+      vm.vmDetails.lastComplianceStatus = (angular.isUndefined(vm.vmDetails.last_compliance_status) ? __('Never Verified') : vm.vmDetails.last_compliance_status);
+      vm.vmDetails.complianceHistory = (vm.vmDetails.compliances.length > 0 ? availableText : notAvailable);
+      var options = {
+        attributes: ['miq_request'],
+      };
+      CollectionsApi.get('services', vmDetails.service.id, options).then(serviceSuccess, {});
```
Comment (AllenBW) [inline client/app/states/vms/details/details.state.js]: why not add this to the state resolve?

File: client/app/states/vms/details/details.state.js [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+    }
+    function defaultText(inputCount, defaultText) {
+      var inputArrSize = inputCount.length;
+      if (defaultText === null) {
+        defaultText = 'None';
+      }
+      if (inputArrSize === 0) {
```
Comment (AllenBW) [inline client/app/states/vms/details/details.state.js]: could be simplified using tertiary

File: client/app/states/vms/details/details.state.js [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+        'security_groups',
+      ];
+      var options = {
+        attributes: requestAttributes,
+      };
+
+      CollectionsApi.get('instances', instanceId, options).then(processInstanceVariables, {});
+    }
+    function processInstanceVariables(data) {
+      var noneText = __('None');
```
Comment (AllenBW) [inline client/app/states/vms/details/details.state.js]: this is already declared in the controller, maybe increase the scope (as it is used in multiple functions)

File: client/app/states/vms/details/details.state.js (hunk 7 above)
Comment (AllenBW) [inline client/app/states/vms/details/details.state.js]: you can pass the result of a resolved item into another item to resolve, same way as you add CollectionApi service

**Comment by chalettu** [comment]
screen1
screen2
sample3

File: client/app/states/vms/details/_details.sass [hunk 10]
```diff
@@ -0,0 +1,25 @@
+.list-view-container
+  &.order-details
+    height: calc(100vh - 272px)
+
+.ss-details-wrapper
+  .order-details
+    .list-view-pf
+      margin-top: 0
+
+
```
Comment (AllenBW) [inline client/app/states/vms/details/_details.sass]: Extra spaceeeeeee

**Comment by chriskacerguis** [comment]
@chalettu can you please link the PT card in here, and then update your PT card?

**Pull request merged**